File: csharp/Main/BirthdayGreeter.cs
using System;
using System.Linq;

namespace Soat.CleanCoders.DipKata.Main
{
    public class BirthdayGreeter
    {
        private readonly FriendRepository _friendRepository;

        public BirthdayGreeter(FriendRepository friendRepository)
        {
            _friendRepository = friendRepository;
        }

        public void SendGreetings()
        {
            var today       = DateTime.Now;
            var emailSender = new EmailSender();
            _friendRepository.FindFriendsBornOn(today)
                             .ToList()
                             .ForEach(friend =>
                             {
                                 var message = mailMessageFor(friend);
                                 emailSender.send(friend.Email, message);
                             });
        }

        private string mailMessageFor(Friend friend) =>
            $"Happy birthday, dear {friend.FirstName}!";
    }
}

File: java/src/main/java/fr/soat/cleancoders/BirthdayGreeter.java
package fr.soat.cleancoders;

import static java.time.MonthDay.now;

public class BirthdayGreeter {
    private final FriendRepository friendRepository;

    public BirthdayGreeter(FriendRepository employeeRepository) {
        this.friendRepository = employeeRepository;
    }

    public void sendGreetings() {
        friendRepository.findFriendsBornOn(now())
                .forEach(friend -> {
                    String message = emailFor(friend);
                    new EmailSender().send(friend.getEmailAdress(), message);
                });
    }

    public String emailFor(Friend friend){
        return String.format("Happy birthday, dear %s!", friend.getName());

    }

}
